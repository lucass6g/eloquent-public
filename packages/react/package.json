{
  "name": "@eloquent/react",
  "version": "1.0.0",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup",
    "clean": "rm -rf dist && rm -rf node_modules"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.1.0",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "react": "^18.3.1",
    "typescript": "^5.4.5"
  }
}
