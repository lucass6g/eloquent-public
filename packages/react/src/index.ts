export {Switch, Case, Default} from './components/conditional-switch'