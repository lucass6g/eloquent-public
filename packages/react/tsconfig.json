{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
