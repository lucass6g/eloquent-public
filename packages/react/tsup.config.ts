import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})