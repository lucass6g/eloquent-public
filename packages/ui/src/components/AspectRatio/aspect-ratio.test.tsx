import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { AspectRatio } from "./index";

describe("AspectRatio", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <AspectRatio ratio={1}>
        <img
          src="https://via.placeholder.com/300/DEDEDE/DEDEDE"
          alt="imagem exemplo"
          className="h-full w-full object-cover rounded-[0.375rem]"
        />
      </AspectRatio>
    );

    expect(container).toMatchSnapshot();
  });

  it("should render correctly", () => {
    const { container } = render(
      <AspectRatio ratio={16 / 9}>
        <img
          src="https://via.placeholder.com/300/DEDEDE/DEDEDE"
          alt="imagem exemplo"
          className="h-full w-full object-cover rounded-[0.375rem]"
        />
      </AspectRatio>
    );

    expect(container.firstChild).toBeInTheDocument();
  });
});