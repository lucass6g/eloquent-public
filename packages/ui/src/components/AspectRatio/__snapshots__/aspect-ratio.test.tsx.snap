// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AspectRatio > should match snapshot 1`] = `
<div>
  <div
    data-radix-aspect-ratio-wrapper=""
    style="position: relative; width: 100%; padding-bottom: 100%;"
  >
    <div
      style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
    >
      <img
        alt="imagem exemplo"
        class="h-full w-full object-cover rounded-[0.375rem]"
        src="https://via.placeholder.com/300/DEDEDE/DEDEDE"
      />
    </div>
  </div>
</div>
`;
